File: Java/labs/lab17_Mediator/task_2_1 – solution/src/ECommerceSite.java
import java.util.HashMap;

public class ECommerceSite {

  private final HashMap<String, Integer> stock = new HashMap<>();

  public ECommerceSite() {
    stock.put("pens", 100);
    stock.put("pencils", 50);
    stock.put("erasers", 75);
  }

  public boolean checkInStock(String item, int quantity) {
    if (stock.containsKey(item) && stock.get(item) > quantity) {
      return true;
    } else {
      return false;
    }
  }

  public void sell(String item, int quantity) {

    int newQuantity = stock.get("pens") - quantity;
    stock.put(item, newQuantity);

  }



}

File: Cpp/lab17_Mediator/task_2_1 - solution/src/ecommerce_site.cpp
#include "ecommerce_site.hpp"


ECommerceSite::ECommerceSite()
{
  this->stock.insert({"pens", 100});
  this->stock.insert({"pencils", 50});
  this->stock.insert({"erasers", 75});
}

bool ECommerceSite::checkInStock(std::string const& item, int quantity)
{
  auto search = this->stock.find(item);
  return search != this->stock.end() && (search->second > quantity);
}

void ECommerceSite::sell(std::string const& item, int quantity)
{
  int newQuantity = this->stock.at(item) - quantity;
  this->stock[item] = newQuantity;
}
